stackit/internal/services/iaas/networkarearoute/resource.go

```diff
@@ -211,21 +211,24 @@ func (r *networkAreaRouteResource) Create(ctx context.Context, req resource.Crea
 
 	var routeId string
 	var route iaas.Route
-	for _, route = range *routes.Items {
-		if !(*route.Prefix == model.Prefix.ValueString() && *route.Nexthop == model.NextHop.ValueString()) {
-			core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating network area route. New static route not found in API response", fmt.Sprintf("Prefix from the route in API: %s", *route.Prefix))
-			return
-		}
 
+	if len(*routes.Items) != 1 {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error creating network area route.", "New static route not found or more than 1 route found in API response.")
+		return
+	}
+
+	// Gets the route ID from the first element, routes.Items[0]
+	for _, route = range *routes.Items {
 		routeId = *route.RouteId
+		break
 	}
```

Comment:
You can get the ID at index 0 using:

```go
routeItems := *routes.Items
routeId := *routeItems[0].RouteId
```

Reply:
oh thx a lot <3 
I needed the route also, so changed to 

	routeItems := *routes.Items
	route := routeItems[0]
	routeId := *route.RouteId